### PR TITLE
Pkg 5219 update v3.0.3

### PR DESCRIPTION
**Pull request by danpetry**
**Destination channel:** defaults

### Links

- [PKG-5219](https://anaconda.atlassian.net/browse/PKG-5219) 
- [Upstream repository](https://github.com/libjpeg-turbo/libjpeg-turbo/tree/3.0.3)
- [Upstream changelog/diff](https://github.com/libjpeg-turbo/libjpeg-turbo/releases)
- Relevant reverse dependency PRs:
  - https://github.com/AnacondaRecipes/torchvision-feedstock/pull/13

### Explanation of changes:

- Required as a dependency for torchvision, their [smoke test](https://github.com/pytorch/vision/commit/1db6907bd137ca006c61657e2a37802e418139d5) checks for it
- Pull changes from conda-forge, keeping changes from our repo that are still relevant
- Pin yasm as per upstream Building.md
- Satisfy linter

[PKG-5219]: https://anaconda.atlassian.net/browse/PKG-5219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 
 package:
   name: libjpeg-turbo
   version: {{ version }}
 
 source:
   url: https://github.com/libjpeg-turbo/libjpeg-turbo/archive/{{ version }}.tar.gz
-  sha256: f892fff427ab3adffc289363eac26d197ce3ccacefe5f5822377348a8166069b
-  # patches:
-  #   - 0001-Exclude-some-tests-when-not-WITH-12BIT.patch
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:11]: Should we delete this patch?

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/bld.bat [hunk 2]
```diff
@@ -1,7 +1,8 @@
 :: Build step
 mkdir build_libjpeg
 cd  build_libjpeg
-cmake -G "NMake Makefiles" ^
+
+cmake -G"Ninja" ^
 	-D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
 	-D CMAKE_BUILD_TYPE=Release ^
 	-D ENABLE_STATIC=1 ^
```
Comment (skupr-anaconda) [inline recipe/bld.bat:8]: Is it fine to have static and shared at the same time?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
-    - jom  # [win]
-    - m2-patch  # [win]
-    - patch     # [not win]
+    # https://github.com/libjpeg-turbo/libjpeg-turbo/blob/7fa4b5b762c9a99b46b0b7838f5fd55071b92ea5/BUILDING.md?plain=1#L16
+    - yasm >=1.2.0
+    - ninja
+  run_constrained:
+    - jpeg <0.0.0a
 
 test:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:28]: I think it is nice to test library files like `libjpeg.so.{version}`, l`ibjpeg.{version}.dylib`, etc. on all supported platforms. See https://github.com/libjpeg-turbo/libjpeg-turbo/blob/3.0.3/BUILDING.md#unx

**Review by skupr-anaconda — APPROVED** [review]
LGTM but testing libraries and binaries would be great to have

File: recipe/meta.yaml (hunk 1 above)
Comment (danpetry) [reply recipe/meta.yaml:11]: deleted

File: recipe/bld.bat (hunk 2 above)
Comment (danpetry) [reply recipe/bld.bat:8]: yep, and it's what we've got already https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Fosx-arm64%2Flibjpeg-turbo-2.0.0-h1a28f6b_0.conda

**Pull request merged**